x-pack/plugins/canvas/canvas_plugin_src/renderers/embeddable/embeddable.tsx

```diff
@@ -105,8 +113,12 @@ export const embeddableRendererFactory = (
           return ReactDOM.unmountComponentAtNode(domNode);
         });
       } else {
-        embeddablesRegistry[uniqueId].updateInput(input);
-        embeddablesRegistry[uniqueId].reload();
+        const embeddable = embeddablesRegistry[uniqueId];
+
+        if ('updateInput' in embeddable) {
```

Comment:
it would be cool if there was a nicer way to check to see if this was a promise or something but it'll work for now

Reply:
Yeah agree with that.  It's probably not a great practice for something to be thing or promise<thing> and instead just let it always be the promise? I dunno but agree it's worth revisiting.